cmake/Modules/Platform/wasi.cmake

```diff
@@ -0,0 +1,3 @@
+set(WASI 1)
+set(UNIX 1)
```

Comment:
WASI currently has a lot in common with UNIX, but it is not a `UNIX` platform per se. Things are still evolving, but right now we're avoiding predefining __unix__ or providing other Unix-specific platform indicators.

Reply:
I think that it is still valuable to define `UNIX`, as this impacts a large surface in CMake.